scripts/setup_irods.py

```diff
@@ -177,8 +212,41 @@ def determine_server_role(irods_config):
         input_filter=irods.lib.set_filter(catalog_service_roles, field='Server role'))
     irods_config.commit(irods_config.server_config, irods_config.server_config_path, clear_cache=False)
 
+def determine_local_storage(irods_config):
+    l = logging.getLogger(__name__)
+    l.info(irods.lib.get_header('Determining local storage'))
+
+    if irods_config.is_provider:
+        default_default = ['yes']
```

Comment:
so, here's the re-runnable issue again.

might need to make a table?

i feel like it's... right?  but not convinced of my own code/thinking.

Reply:
I probably just don't know what this is going to look like. The way I'm thinking of it, here's what I'm going to see at setup for a provider:
```
Local storage on this server [yes]:
```
And then here's what I will see for a consumer:
```
Local storage on this server [no]:
```
Is that correct? If not, maybe a paste of what this will look like to someone setting up a server would be good.

Reply:
I believe you are reading/parsing it correctly - this is what it will currently do.

And I also think that default_default should probably be 'yes' for consumer as well on first run.

Definitely need a table.

Reply:
it will go to... the default resource... which must be elsewhere.

so... i think it should still just work?

Reply:
A lone provider, with no local storage, will definitely not 'just work'.   And that's okay - that's a broken installation attempt, I feel.   And the error should alert the administrator that they're doing it wrong.

Reply:
I think so, yes.

Reply:
Or even just remove the idea of default_default... since there's no longer any variation?

Reply:
Please take a look at the logic for this.

I changed it so that both providers and consumers default to yes for local storage.

Reply:
Yes, I think that's correct.

Default is to have local storage, for all servers (well, except maybe 'proxy' role, which doesn't exist yet).

Reply:
Ok. Marking as resolved.